src/batch/src/executor/insert.rs

```diff
@@ -87,6 +95,23 @@ impl InsertExecutor {
             sorted_default_columns,
             row_id_index,
             returning,
+            txn_id,
+            write_handle: None,
+            finished: false,
+        }
+    }
+}
+
+impl Drop for InsertExecutor {
+    fn drop(&mut self) {
+        // If this executor is not finished, try to rollback.
+        if !self.finished {
+            if let Some(write_handle) = self.write_handle.take() {
+                let txn_id = self.txn_id;
+                tokio::spawn(async move {
+                    let _ = write_handle.write_txn_msg(TxnMsg::Rollback(txn_id)).await;
+                });
+            }
```

Comment:
It is possible that users cancel a running DML which could cause a dangling transaction, so we need to send `Rollback` message to the downstream. It seems not a best practice to spawn a task on `drop` method, but it is the easiest way to rollback a transaction if an unexpected interruption occurs.

Reply:
I'm considering whether we can implement `Drop` on `WriteHandle` and maintain the transaction ID and `finished` state there, so that there's no need to implement for all DML executors.